File: csharp/DiagramPrinter/FlowchartDiagram.cs
using System.Security.Authentication;

namespace DiagramPrinter;

/**
 * This is the production class used for real.
 * The implementation is largely omitted here, imagine it is much bigger and more complex.
 */
public class FlowchartDiagram
{
    public FlowchartDiagram()
    {
        throw new AuthenticationException("you can't construct this in a unit test");
    }

    public string Name()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public string SerialNumber()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public bool IsDisposed()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public PdfDocument FlowchartAsPdf()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public SpreadsheetDocument FlowchartDataAsSpreadsheet()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public string SummaryInformation()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }

    public PngDocument FlowchartThumbnail()
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }
    
    public FlowchartReport Report()
    {
        throw new NotImplementedException();
    }

    public FlowchartDiagram Summary()
    {
        throw new NotImplementedException();
    }

    public FlowchartReportItems ValidationProblems(string reportTemplate, List<string> data)
    {
        throw new NotImplementedException();
    }

    public FlowchartReportItems ReportData()
    {
        return new FlowchartReportItems(this.Name(), this.SerialNumber(), this.FlowchartThumbnail().Filename());
    }

    public List<DiagramPage> PagesData()
    {
        // fake implementation
        return new List<DiagramPage>()
        {
            new DiagramPage(this.Name(), "page 1"),
            new DiagramPage(this.Name(), "page 2"),
        };
    }
    // Imagine about 200 more methods here
}

public class FlowchartReport
{
    public void Close()
    {
        throw new NotImplementedException();
    }

    public void Open(bool show)
    {
        throw new NotImplementedException();
    }

    public bool isOpen()
    {
        throw new NotImplementedException();
    }

    public void OpenWithContents(string reportTemplate, FlowchartReportItems substitutions, bool readOnly)
    {
        throw new NotImplementedException();
    }

    public void SaveToFile(string targetFilename)
    {
        throw new NotImplementedException();
    }
}

File: java/src/main/java/sammancoaching/FlowchartDiagram.java
package sammancoaching;

import javax.naming.AuthenticationException;
import java.util.ArrayList;
import java.util.List;

public class FlowchartDiagram {

    public FlowchartDiagram() throws AuthenticationException {
        throw new RuntimeException("You can't construct this in a unit test");
    }

    public String getName() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public String getSerialNumber() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public boolean isDisposed() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public PdfDocument getFlowchartAsPdf() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public SpreadsheetDocument getFlowchartDataAsSpreadsheet() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public String getSummaryInformation() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public PngDocument getFlowchartThumbnail() {

        throw new RuntimeException("Can't call this from a unit test");
    }

    public FlowchartReport report() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public FlowchartDiagram summary() {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public FlowchartReportItems ValidationProblems(String reportTemplate, List<String> data) {
        throw new RuntimeException("Can't call this from a unit test");
    }

    public FlowchartReportItems reportData() {
        return new FlowchartReportItems(this.getName(), this.getSerialNumber(), this.getFlowchartThumbnail().getFilename());
    }

    public List<DiagramPage> PagesData() {
        // fake implementation
        var list = new ArrayList<DiagramPage>();
        list.add(new DiagramPage(this.getName(), "page 1"));
        list.add(new DiagramPage(this.getName(), "page 2"));
        return list;
    }
}
